assets/javascripts/discourse/services/chat.js

```diff
@@ -650,6 +650,7 @@ export default Service.extend({
         dmChatChannel.set("last_message_sent_at", new Date());
         this.reSortDirectMessageChannels();
       }
+      this.appEvents.trigger("chat:refresh-channels");
```

Comment:
Could you elaborate more on these two please?

Reply:
This one is observing `this.messageBus.subscribe(`/chat/${channel.id}/new-messages`, (busData) => {`

When a new message is available in chat, I trigger refresh channels. It will update sidebar to show "unread" indicator